File: HomeAutomation/HomeAutomation/HomeAutomation/Models/Command.cs
namespace HomeAutomation.Models;

public class Command
{
    private string command;
    private float value;

    public Command(string command, float value)
    {
        this.command = command;
        this.value = value;
    }

    public Command(string command)
    {
        this.command = command;
        this.value = 0;
    }

    public Command(float value)
    {
        this.command = "";
        this.value = value;
    }
    
    public string getCommand()
    {
        return command;
    }
    
    public float getValue()
    {
        return value;
    }
    
}

File: TaskManager/task-manager-lab-final/src/main/java/commands/Command.java
package commands;

import models.CommandType;

import java.security.Timestamp;

public abstract class Command {
    String commandEntry;
    String description;
    CommandType commandType;
    protected Timestamp timestamp;
    abstract public void execute();

    abstract public void undo();

    public Command(String issue, CommandType type){
        this.commandEntry = issue;
        this.commandType = type;
    }

    public Timestamp getTimestamp() {
        return timestamp;
    }

    public String getDescription() {
        return description;
    }

    public CommandType getType() {
        return commandType;
    }
    
}
